### PR TITLE
planned vision

### PR DESCRIPTION
**Pull request by Opeyem1a**

### CODE REVIEW

File: VISION.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+   partial string and then menu select a branch to hop to, basically git switch
+   with better search & UI
+2. `continue` - basically just `git rebase --continue`
+3. `all` - basically just `git add .`
+4. `onto` - opens a menu to select a branch, rebases your current branch "onto"
+   that chosen branch
+5. `track` - start "tracking" current branch
+6. `go` / `send` (idk what to call this) - basically `git commit -am ""` and
+   takes a message argument
+7. `feat` / `create` - basically `git checkout -b [name] && git commit -am ""`
```
Comment (aadiraju) [inline VISION.md:40]: maybe gum spawn

File: VISION.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+   where the branch name is auto-generated from the commit message
+8. `ls` - view tracked branches in a tree structure visually
```
Comment (aadiraju) [inline VISION.md:42]: so this is just hop without actually hopping?

Comment (Opeyem1a) [reply VISION.md:42]: It would show the tree structure though, like this:
image

File: VISION.md (hunk 1 above)
Comment (Opeyem1a) [reply VISION.md:40]: Maybe

Comment (Opeyem1a) [reply VISION.md:40]: Maybe just `new`

File: VISION.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+1. `hop` - list all branches, not just tracked ones, let's you search with
+   partial string and then menu select a branch to hop to, basically git switch
+   with better search & UI
+2. `continue` - basically just `git rebase --continue`
+3. `all` - basically just `git add .`
+4. `onto` - opens a menu to select a branch, rebases your current branch "onto"
+   that chosen branch
+5. `track` - start "tracking" current branch
+6. `go` / `send` (idk what to call this) - basically `git commit -am ""` and
```
Comment (MtheDV) [inline VISION.md:38]: Not sure either, but if it's just committing files, then `go` or `send` feels off? Could be something around prepping for commit? `prep`, `ready`, `hold`, `add`?

**Review by MtheDV — COMMENTED** [review]
I think this looks good so far, don't have too much to say about it. Timeline seems reasonable. Do you think we would want something around merging? (I use that more than rebase)

File: VISION.md (hunk 3 above)
Comment (Opeyem1a) [reply VISION.md]: Made a new structure for the command names. Idea is to have them be memorable words and then just have abbreviated aliases

File: VISION.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+### Commands that require we build "the tree"
+
+The tree, meaning the way we internally store the representation of which
+branches are stacked on which branches.
+
+-   `sync`, `up`, `down`, `ls`, `track`
+
+### Commands that can be built now, might need to be changed after "the tree"
+
+-   `brnach attach`, `continue`
```
Comment (MtheDV) [inline VISION.md:71]: brnach lol

Comment (Opeyem1a) [reply VISION.md]: Oops

**Pull request merged**